src/library/librarycontrol.cpp

```diff
@@ -482,16 +482,26 @@ void LibraryControl::emitKeyEvent(QKeyEvent&& event) {
     VERIFY_OR_DEBUG_ASSERT(m_pLibraryWidget) {
         return;
     }
-    if (!m_pLibraryWidget->hasFocus() && !m_pSidebarWidget->hasFocus()) {
+    if (!QApplication::focusWindow()) {
+        qDebug()<<"Mixxx window is not focused, don't send key events";
+    }
+
+    Qt::Key key = static_cast<Qt::Key>(event.key());
+//    qDebug() << "|   key  " << key;
+
+    // If the main window has focus, any widget can receive Tab.
+    // Other keys should be sent to library widgets only to not
+    // accidentally alter spinboxes etc.
+    if (key != Qt::Key_Tab && !m_pSidebarWidget->hasFocus()
+            && !m_pLibraryWidget->getActiveView()->hasFocus()) {
         setLibraryFocus();
     }
```

Comment:
do you want return her as well?

Reply:
nope. this is meant to catch Up / Down / L / R moves which are executed below.